runtime/gc_structs/ClassIterator.cpp

```diff
@@ -70,8 +70,13 @@ GC_ClassIterator::nextSlot()
 		_state += 1;
 
 	case classiterator_state_slots:
-		if (slotOffsets[_scanIndex] != 0) {
-			return (j9object_t *)((U_8*)_clazzPtr + (UDATA)slotOffsets[_scanIndex++]);	
+		while (slotOffsets[_scanIndex] != 0) {
+			//shouldScanClassObject is true by default, in the case of balanced GC check if object is ClassObject
```

Comment:
comment with "//" is not allowed, please use "/*...*/"

Reply:
👍